test/games/strategy/engine/data/UnitCollectionTest.java

```diff
@@ -0,0 +1,386 @@
+    final IntegerMap<UnitType> unitsByType = uc.getUnitsByType(defaultPlayerID);
+    assertThat(unitsByType.getInt(unitTypeOne), is(equalTo(unitCountDefaultPlayerUnitTypeOne)));
+    assertThat(unitsByType.getInt(unitTypeTwo), is(equalTo(unitCountDefaultPlayerUnitTypeTwo)));
+  }
+
+  @Test
+  public void getUnitsWithUnityByTypeIntegerMap() {
+    addAllDefaultPlayerUnits();
+    final IntegerMap<UnitType> unitsByType = uc.getUnitsByType();
+    final Collection<Unit> expAllUnitsOfDefaultPlayer = uc.getUnits(unitsByType);
+    assertThat(expAllUnitsOfDefaultPlayer.size(), is(equalTo(unitCountDefaultPlayer)));
+  }
+
+  @Test
+  public void size() {
+    assertThat(uc.size(), is(equalTo(0)));
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeOne());
+    assertThat(uc.size(), is(equalTo(unitCountDefaultPlayerUnitTypeOne)));
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeTwo());
+    assertThat(uc.size(), is(equalTo(unitCountDefaultPlayerUnitTypeOne + unitCountDefaultPlayerUnitTypeTwo)));
+  }
+
+  @Test
+  public void isEmpty() {
+    assertThat(uc.isEmpty(), is(equalTo(true)));
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeOne());
+    assertThat(uc.isEmpty(), is(equalTo(false)));
+  }
+
+  @Test
+  public void getUnits() {
+    final Collection<Unit> unitsOfOtherPlayerOfUnitTypeOne = getUnitsOfOtherPlayerOfUnitTypeOne();
+    uc.addAllUnits(unitsOfOtherPlayerOfUnitTypeOne);
+    assertThat(uc.getUnits().containsAll(unitsOfOtherPlayerOfUnitTypeOne), is(equalTo(true)));
+  }
+
+  @Test
+  public void getPlayersWithUnits() {
+    assertThat(uc.getPlayersWithUnits().size(), is(equalTo(0)));
+    addAllDefaultPlayerUnits();
+    assertThat(uc.getPlayersWithUnits().size(), is(equalTo(1)));
+    addAllOtherPlayerUnits();
+    assertThat(uc.getPlayersWithUnits().size(), is(equalTo(2)));
+  }
+
+  @Test
+  public void getPlayerUnitCounts() {
+    addAllPlayerUnits();
+    final IntegerMap<PlayerID> playerUnitCounts = uc.getPlayerUnitCounts();
+    assertThat(playerUnitCounts.getInt(defaultPlayerID), is(equalTo(unitCountDefaultPlayer)));
+    assertThat(playerUnitCounts.getInt(otherPlayerID), is(equalTo(unitCountOtherPlayer)));
+  }
+
+  @Test
+  public void hasUnitsFromMultiplePlayers() {
+    addAllDefaultPlayerUnits();
+    assertThat(uc.hasUnitsFromMultiplePlayers(), is(equalTo(false)));
+    addAllOtherPlayerUnits();
+    assertThat(uc.hasUnitsFromMultiplePlayers(), is(equalTo(true)));
+  }
+
+  @Test
+  public void getHolder() {
+    assertThat(uc.getHolder(), is(equalTo(defaultPlayerID)));
+  }
+
+  @Test
+  public void allMatch() {
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeOne());
+    assertThat(uc.allMatch(Matches.unitIsOfType(unitTypeOne)), is(equalTo(true)));
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeTwo());
+    assertThat(uc.allMatch(Matches.unitIsOfType(unitTypeOne)), is(equalTo(false)));
+  }
+
+  @Test
+  public void someMatch() {
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeOne());

```

Comment:
The `UC` setup should just be a few different final variables you set up in 'setup'

```
UnitCollection defaultPlayerTypeOneCollection = getUnitsOfDefaultPlayerOfUnitTypeOne();
```

Looks like you'll have 3 or 4 of them, and lots of common set up code would drop away.